linux/LinuxProcessList.c

```diff
@@ -108,28 +85,14 @@ static int sortTtyDrivers(const void* va, const void* vb) {
 
 static void LinuxProcessList_initTtyDrivers(LinuxProcessList* this) {
    TtyDriver* ttyDrivers;
-   int fd = open(PROCTTYDRIVERSFILE, O_RDONLY);
-   if (fd == -1)
-      return;
 
-   char* buf = NULL;
-   int bufSize = MAX_READ;
-   int bufLen = 0;
-   for (;;) {
-      buf = xRealloc(buf, bufSize);
-      int size = xread(fd, buf + bufLen, MAX_READ);
-      if (size <= 0) {
-         buf[bufLen] = '\0';
-         close(fd);
-         break;
-      }
-      bufLen += size;
-      bufSize += MAX_READ;
-   }
-   if (bufLen == 0) {
-      free(buf);
+   char buf[16384];
+   ssize_t r = xReadfile(PROCTTYDRIVERSFILE, buf, sizeof(buf) - 1);
+   if (r < 0)
       return;
-   }
+
+   buf[r] = '\0';
```

Comment:
What about making the NUL termination part of the xReadfile API? I.e. internally ensured before terminating that the byte following the last index written in buffer is forced to `\0`?